universal-application-tool-0.0.1/app/repository/VersionRepository.java

```diff
@@ -57,56 +59,48 @@ public void publishNewSynchronizedVersion() {
       Version active = getActiveVersion();
       Preconditions.checkState(
           draft.getPrograms().size() > 0, "Must have at least 1 program in the draft version.");
+
+      ImmutableSet<String> draftProgramsNames =
+          draft.getPrograms().stream()
+              .map(program -> program.getProgramDefinition().adminName())
+              .collect(ImmutableSet.toImmutableSet());
+
+      ImmutableSet<String> draftQuestionNames =
+          draft.getQuestions().stream()
+              .map(question -> question.getQuestionDefinition().getName())
+              .collect(ImmutableSet.toImmutableSet());
```

Comment:
suggestion (non-blocking): could make these methods `Version#getProgramNames` and `Version#getQuestionNames`

Reply:
done!